lib/console.js

```diff
@@ -49,9 +55,32 @@ Console.prototype.error = Console.prototype.warn;
 
 
 Console.prototype.dir = function(object, options) {
-  this._stdout.write(util.inspect(object, util._extend({
-    customInspect: false
-  }, options)) + '\n');
+  this._stdout.write((this._group ? util.inspect(object, util._extend({
+        customInspect: false }, options)).replace(/^/gm, this._prefix) :
+          util.inspect(object, util._extend({ customInspect: false },
+              options)).replace(/^/gm, this._prefix)) + '\n');
+};
+
+
+Console.prototype.group = function() {
+  if (arguments.length) {
+    this.log.apply(this, arguments);
+  }
+  this._group++;
+  this._prefix = '  '.repeat(this._group);
+};
+
+
+Console.prototype.groupCollapsed = function() {
+  this.group.apply(this, arguments);
+};
+
+
+Console.prototype.groupEnd = function() {
+  if ( this._group > 0 ) {
+    this._group--;
+    this._prefix = (this._group ? '  '.repeat(this._group) : '');
```

Comment:
Tried to avoid extra function call and use single `''` const that optimizes better than generating a new one for each `groupEnd`.

If you feel complicates code too much, this conditional can be removed as you suggested.